akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -207,7 +207,7 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
           super.aroundPreRestart(reason, Some(m))
         case mo ⇒
           flushJournalBatch()
-          super.aroundPreRestart(reason, None)
+          super.aroundPreRestart(reason, mo)
```

Comment:
@drewhk @naferx @patriknw 
I removed the type match from this clause, so this PR is ready to merge.